pandas/tests/io/test_sql.py

```diff
@@ -902,6 +902,36 @@ def test_to_sql_replace(self, test_frame1):
 
         assert num_rows == num_entries
 
+    def test_to_sql_truncate(self, test_frame1):
+        sql.to_sql(test_frame1, "test_frame3", self.conn, if_exists="fail")
+        # Add to table again
+        sql.to_sql(test_frame1, "test_frame3", self.conn, if_exists="truncate")
+        assert sql.has_table("test_frame3", self.conn)
+
+        num_entries = len(test_frame1)
+        num_rows = count_rows(self.conn, "test_frame3")
+
+        assert num_rows == num_entries
+
+    def test_to_sql_truncate_no_table(self, test_frame1):
+        #  creates new table if table doesn't exist
+        sql.to_sql(test_frame1, "test_frame_new", self.conn, if_exists="truncate")
```

Comment:
Think this should raise too? Feels a little strange to have truncate delete a table

Reply:
Though I guess this is consistent with `replace`